pykwasm/src/pykwasm/kdist/wasm-semantics/kwasm-lemmas.md

```diff
@@ -267,13 +267,66 @@ In this case, it's simpler (and safe) to simply discard the `#chop`, instead of
     rule #minSigned(ITYPE) => 0 -Int #pow1(ITYPE)
 ```
 
-Lookups
--------
+Map Lookups
+-----------
 
 ```k
     rule (_MAP:Map [KEY <- VAL])[KEY] => VAL [simplification]
 ```
 
+List Operations
+---------------
+
+These should probably be integrated into `domains.md`
+
+```k
+    rule (L1:List ListItem(X:KItem) _:List)[size(L1)] => X
+      [simplification]
+
+    rule (L1:List ListItem(_:KItem) L2:List)[size(L1) <- V] =>
+      L1:List ListItem(V) L2:List
+      [simplification]
```

Comment:
Is the Haskell backend able to use these two rules?

Reply:
Yes, it' using them for the `<locals>` and `<mem>` specs: https://github.com/Pi-Squared-Inc/wasm-semantics/pull/40/files#diff-56e769c3ddbacd55039dc6c309d8126d4f9d67c1ed9cbaccdf22ac284aa971ebR8

Unfortunately, not for the `wrc-20` spec. I assume the issue is that while it works as an initial configuration, it cannot be matched as a final configuration; that is why the `locals-loop` spec also only works when the accessed local is the last in the list.